app/views/posts/_list.haml

```diff
@@ -28,7 +28,9 @@
         %th.sub Unread
       %th.sub Last Updated
       - if check_box_name
-        %th.sub
+        %th.sub.check-all-box
+          - unless posts.empty?
+            = check_box_tag :check_all, check_box_name, false, class: 'check-all', id: 'check-all'
```

Comment:
If there's multiple post lists on a page that have checkboxes, this might cause invalid HTML (by duplicating the ID). I think most post lists don't have checkboxes, so maybe this isn't an issue - but can you think of anywhere that might be impacted?

(Is the ID necessary? It seems like you did some work to avoid needing an ID by specifying the corresponding `name` for each checkbox, so we might just be able to remove `id` from each of these definitions)